src/connectors/exchangeRate/index.ts

```diff
@@ -0,0 +1,251 @@
+import axios from 'axios'
+
+import { CACHE_TTL } from 'common/enums'
+import { environment } from 'common/environment'
+import { NetworkError, UnknownError } from 'common/errors'
+import logger from 'common/logger'
+import { CacheService } from 'connectors'
+import {
+  GQLExchangeRate,
+  GQLQuoteCurrency,
+  GQLTransactionCurrency,
+} from 'definitions'
+
+// TYPES
+
+interface Pair {
+  from: GQLTransactionCurrency
+  to: GQLQuoteCurrency
+}
+
+type TokenCurrency = GQLTransactionCurrency.LIKE | GQLTransactionCurrency.USDT
+type FiatCurrency = GQLTransactionCurrency.HKD
+
+// CONSTANTS
+
+const tokenCurrencies: TokenCurrency[] = [
+  'LIKE' as GQLTransactionCurrency.LIKE,
+  'USDT' as GQLTransactionCurrency.USDT,
+]
+const fiatCurrencies: FiatCurrency[] = ['HKD' as GQLTransactionCurrency.HKD]
+const quoteCurrencies: GQLQuoteCurrency[] = [
+  'TWD' as GQLQuoteCurrency.TWD,
+  'HKD' as GQLQuoteCurrency.HKD,
+  'USD' as GQLQuoteCurrency.USD,
+]
+
+const TOKEN_TO_COINGECKO_ID = {
+  LIKE: 'likecoin',
+  USDT: 'tether',
+} as const
+
+const COINGECKO_API_URL = 'https://api.coingecko.com/api/v3/simple/price'
+const EXCHANGE_RATES_DATA_API_URL =
+  'https://api.apilayer.com/exchangerates_data/latest'
```

Comment:
No API versioning? It seems easy to break if any changes on API Layer.

Reply:
It seem that many exchange rates api do not use API versioning, like Open Exchange Rates API .  When those APIs change,  queue cron job will fail and send alert to Slack,  at the same time server still return old rates data in redis (TTL 10 days now). Hence,  I think we have enough time to response in the case of API breaking changing